### DISCUSSION TITLE
Should I use node termination handler (NTH) for EKS managed node group

### DISCUSSION THREAD

**Discussion by chrisp018** [aws/aws-node-termination-handler#559] [Q&A] [answered]
- As this[ link ](https://aws.amazon.com/vi/blogs/containers/amazon-eks-now-supports-provisioning-and-managing-ec2-spot-instances-in-managed-node-groups/) states: To handle Spot interruptions, you do not need to install any extra automation tools on the cluster, for example, AWS Node Termination Handler (EKS managed node group) => do we need to implement NTH in the managed node group?
- Do the steps of handling interruption in managed group graceful?  _"When a replacement Spot node is bootstrapped and in the Ready state on Kubernetes, Amazon EKS cordons and drains the Spot node that received the rebalance recommendation. Cordoning the Spot node ensures that the service controller doesn't send any new requests to this Spot node. It also removes it from its list of healthy, active Spot nodes. Draining the Spot node ensures that running pods are evicted gracefully."_ [Link](https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html#managed-node-group-capacity-types). 
because it will create a new spot instance and once the instance is in a ready state -> the pod will be re-deployed in the new ready instance. It can cause service downtime when the new instance takes more time to be bootstrapped and in a ready state (the spot already reclaimed before the new instance is in a ready state). In this case, should I implement NTH for managed instances to allow pods evicted right after receiving rebalance recommendations?

**Comment by bwagner5** [comment, marked as answer]
EKS Managed Node Groups already gracefully cordons and drains nodes. So there is no need to use NTH if you are only using Managed Node Groups.
